src/Compilers/Test/Utilities/CSharp/UsesIsNullableVisitor.cs

```diff
@@ -0,0 +1,173 @@
+
+        internal static void GetUses(ArrayBuilder<Symbol> builder, Symbol symbol)
+        {
+            var visitor = new UsesIsNullableVisitor(builder);
+            visitor.Visit(symbol);
+            visitor.Free();
+        }
+
+        public override bool VisitNamespace(NamespaceSymbol symbol)
+        {
+            return VisitList(symbol.GetMembers());
+        }
+
+        public override bool VisitNamedType(NamedTypeSymbol symbol)
+        {
+            // PROTOTYPE(NullableReferenceTypes): Check BaseType and Interfaces type arguments.
+            if (AddIfUsesIsNullable(symbol, symbol.TypeParameters))
+            {
+                return true;
+            }
+            return VisitList(symbol.GetMembers());
+        }
+
+        public override bool VisitMethod(MethodSymbol symbol)
+        {
+            return AddIfUsesIsNullable(symbol, symbol.TypeParameters) ||
+                AddIfUsesIsNullable(symbol, symbol.ReturnType) ||
+                VisitList(symbol.Parameters);
+        }
+
+        public override bool VisitProperty(PropertySymbol symbol)
+        {
+            return AddIfUsesIsNullable(symbol, symbol.Type) ||
+                VisitList(symbol.Parameters);
+        }
+
+        public override bool VisitEvent(EventSymbol symbol)
+        {
+            return AddIfUsesIsNullable(symbol, symbol.Type);
+        }
+
+        public override bool VisitField(FieldSymbol symbol)
+        {
+            return AddIfUsesIsNullable(symbol, symbol.Type);
+        }
+
+        private bool VisitList<TSymbol>(ImmutableArray<TSymbol> symbols) where TSymbol : Symbol
+        {
+            bool result = false;
+            foreach (var symbol in symbols)
+            {
+                if (this.Visit(symbol))
+                {
+                    result = true;
+                }
+            }
+            return result;
+        }
+
+        private bool AddIfUsesIsNullable(Symbol symbol, ImmutableArray<TypeParameterSymbol> typeParameters)
+        {
+            foreach (var type in typeParameters)
+            {
+                if (UsesIsNullable(type))
+                {
+                    _builder.Add(symbol);
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        private bool AddIfUsesIsNullable(Symbol symbol, TypeSymbolWithAnnotations type)
+        {
+            if (UsesIsNullable(type))
+            {
+                _builder.Add(symbol);
+                return true;
+            }
+            return false;
+        }
+
+        private bool UsesIsNullable(TypeSymbolWithAnnotations type)
+        {
+            if (type is null)
+            {
+                return false;
+            }
+            var typeSymbol = type.TypeSymbol;
+            return (type.IsNullable != null && type.IsReferenceType) ||
+                UsesIsNullable(type.TypeSymbol);
+        }
+
+        // Should use TypeSymbolExtensions.VisitType if/when that
```

Comment:
PROTOTYPE

Reply:
It's not a requirement that the test helper use 'VisitType' since it may be useful to have a distinct implementation here, because the requirements here may be different, or to catch changes to 'VisitType'. I've removed the comment.

---
In reply to: [171084332](https://github.com/dotnet/roslyn/pull/25087#discussion_r171084332) [](ancestors = 171084332)